contrib/scalar/scalar.c

```diff
@@ -19,9 +19,27 @@ static int is_unattended(void) {
 	return git_env_bool("Scalar_UNATTENDED", 0);
 }
 
+static void strbuf_parentdir(struct strbuf *buf)
+{
+	int len = buf->len;
+	while (len > 0 && !is_dir_sep(buf->buf[--len]))
+		; /* keep looking for parent directory */
+	strbuf_setlen(buf, len);
+}
+
+/* Given an absolute working directory, find the enlistment root */
+static void strbuf_enlistmentroot(struct strbuf *buf)
```

Comment:
Could I ask for a `_` between `enlistment` and `root`? That will shut up my VS Code's cSpell... ;-)

Reply:
I think this is is out-of-date - the function was removed in favor of `find_enlistment()` (+other general refactors). 

As a note, I had separated the original commit & `fixup!` commit that included changes based on review comments to make it easier to review the implementation of those changes. Since (I think) those have been addressed/re-reviewed, I've squashed everything together into a single commit again.

Reply:
I think we should use the form `fixup!` in the final commit, so that the faulty commit in our branch thicket can be fixed in the rebase to the next Git for Windows version.

Reply:
That works for me - once all other review comments are addressed, I'll restructure the commit history into a series of fixups.

Reply:
I've restructured the commit history, applying fixups to (I think) the appropriate commits. I'll also be able to help with any merge conflict resolution later on (based on my local rebase, it's pretty messy due to the context differences).